presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -104,10 +107,9 @@
     protected final String connectionUrl;
     protected final Properties connectionProperties;
     protected final String identifierQuote;
-    private final boolean mapTableNames;
 
-    private Map<String, Map<String, String>> schemaTableMapping = new HashMap<>();
-    private Map<String, ReentrantLock> schemaTableMappingLock = new ConcurrentHashMap<>();
+    private final LoadingCache<String, Optional<String>> schemaMappingCache;
+    private final Map<String, LoadingCache<String, Optional<String>>> schemaTableMapping = new ConcurrentHashMap<>();
```

Comment:
Not sure why you're asking to hide these fields as a separate class. Something like a SchemaTableMapping class? Pass in a BaseJdbcClient object to the constructor, and use that to perform all of the loading? Problem with this is how would I get access to the raw RDBMS schema/table names? The public API (currently) only provides for the lowercased names).

Hmm...I suppose I could instead create an implementation of the BaseJdbcClient in the test context which doesn't actually connect to JDBC and returns pre-defined lists of tables/schemas.